crypto/evp/evp_enc.c

```diff
@@ -913,11 +914,11 @@ int EVP_DecryptFinal_ex(EVP_CIPHER_CTX *ctx, unsigned char *out, int *outl)
 
     /*
      * Check if errors occured while decrypting the final block.
-     * Note that ERR_raise is not called.
+     * Note that ERR_raise is not called if ret != 0 and soutl > INT_MAX.
      */
-    int ret_nz = ~constant_time_eq_int(ret, 0);
-    int soutl_gt = constant_time_lt_s(INT_MAX, soutl);
-    ret &= ~soutl_gt;
+    unsigned int ret_nz = ~constant_time_eq_int(ret, 0);
+    size_t int_max_soutl_lt = constant_time_lt_s(INT_MAX, soutl);
```

Comment:
Detecting a buffer that's been padded from less to greater than INT_MAX doesn't seem like much of a weakness but better to fix it (as you've done).